packages/bot/src/stores/bot-store.js

```diff
@@ -1,10 +1,53 @@
-import { observable } from 'mobx';
+import { action, reaction } from 'mobx';
 
 export default class BotStore {
-    @observable title = 'Hi new bot!';
+    constructor(root_store) {
+        this.root_store = root_store;
+    }
 
-    /* @action.bound onClick = () => {
-        // eslint-disable-next-line
-        this.ws.subscribeTicks('R_50', t => console.log(t), false);
-    } */
+    @action.bound
+    registerAccountSwitcherListener() {
+        const { client } = this.root_store.core;
+        const { active_symbols, contracts_for } = this.root_store.api_helpers;
+
+        this.switchAccountDisposer = reaction(
+            () => client.switch_broadcast,
+            (switch_broadcast) => {
+                if (!switch_broadcast) {
+                    return;
+                }
+
+                const { derivWorkspace: workspace } = Blockly;
+
+                if (workspace) {
+                    active_symbols.retrieveActiveSymbols(true).then(() => {
+                        contracts_for.disposeCache();
+                        workspace.getAllBlocks()
+                            .filter(block => block.type === 'trade_definition_market')
+                            .forEach(block => {
+                                const fake_create_event = new Blockly.Events.Create(block);
+                                Blockly.Events.fire(fake_create_event);
+                            });
+                    });
+                }
+            }
+        );
+    }
+
+    @action.bound
+    disposeSwitchAccount() {
+        if (typeof this.switchAccountDisposer === 'function') {
+            this.switchAccountDisposer();
+        }
+    }
+
+    @action.bound
+    onMount() {
+        this.registerAccountSwitcherListener();
+    }
+
+    @action.bound
+    onUnmount() {
+        this.disposeSwitchAccount();
```

Comment:
@aaron-binary if your not calling `disposeSwitchAccount` anywhere else, you can remove it and add the code to unmount itself.

Reply:
I understand what you're saying but this would make it more consistent with trader (they're disposing of their concerns in the same way (see e.g. `base-store` or `trade-store`). Later when we add more things that need disposing we'll add this code again.